src/activity_queue.rs

```diff
@@ -220,27 +201,145 @@ pub(crate) fn generate_request_headers(inbox_url: &Url) -> HeaderMap {
     headers
 }
 
+/// A simple activity queue which spawns tokio workers to send out requests
+/// When creating a queue, it will spawn a task per worker thread
+/// Uses an unbounded mpsc queue for communication (i.e, all messages are in memory)
+pub(crate) struct ActivityQueue {
+    // Our "background" tasks
+    senders: Vec<UnboundedSender<SendActivityTask>>,
+    // Round robin of the sender list
+    last_sender_idx: AtomicUsize,
+    // Stats shared between the queue and workers
+    stats: Arc<Stats>,
+}
+
+/// Simple stat counter to show where we're up to with sending messages
+/// This is a lock-free way to share things between tasks
+/// When reading these values it's possible (but extremely unlikely) to get stale data if a worker task is in the middle of transitioning
+#[derive(Default)]
+struct Stats {
+    pending: AtomicUsize,
+    running: AtomicUsize,
+    dead: AtomicUsize,
+    complete: AtomicUsize,
+}
+
+/// We need to retry activity sending in case the target instances is temporarily unreachable.
+/// In this case, the task is stored and resent when the instance is hopefully back up. This
+/// list shows the retry intervals, and which events of the target instance can be covered:
+/// - 60s (one minute, service restart)
+/// - 60min (one hour, instance maintenance)
+/// - 60h (2.5 days, major incident with rebuild from backup)
+/// TODO: make the intervals configurable
```

Comment:
I dont think this needs to be configurable, its better to provide a sensible default. But if you need this for some reason, open an issue and provide the reasons there instead of leaving this comment.

Reply:
This was just the existing comment carried across. I will remove the TODO

Reply:
Sorry I thought it was yours.